Fix homepage test

kawrou opened the pull request:
### Fixed Tests:
1. pages/HomePage
2. services/RecipeScraper
3. services/authentication (checkToken)

### Description:
I worked from services back up to home page. 

**services/authentication (checkToken) & api/controllers/authentication:**

CheckToken seemed integral to the functioning of RecipeScraper so I started with that and refactored the code a little bit. 
As CheckToken only requires that the controller sends a response code, I changed the controller to send 200 or 500. 
401 is handled by tokenChecker. 
I got rid of the data in services as the data isn't used for anything.
**Please check that I'm correct in that assumption**
Left a comment on the controller to remind us that a test needs to be written for /controller/authentication (checkToken)

Still unsure about:
`return { token: data.token, user_id: data.user_id };`
user_id doesn't contain any data as the controller doesn't give back any. 
Seemed to work fine when I got rid of it despite when Mat tried it. 

**components/RecipeScraper**:
Changed window.localStorage.getItem() to use token as prop instead. App seems to work fine with the change. 
Left a comment to show it. 

Atm: Click "Generate Recipe" with empty URL -> Create Recipe Page
Is this ok? 

Added an aria-label to a `<div>` for accessibility and testing. But then didn't use it for testing.
`return (
    <div className="w-full pt-5">
    <div className="w-full pt-5" aria-label="Recipe Url Form">
      <input
        type="text"
        value={url}`

**I wonder if instead of a div, it should be a `<form>` for accessibility. Thoughts??**


### TESTS:

Fixed test for services/authentication:
- login (left comment about user_id)
- checkToken - Tests status 200 and Error

Fixed test for RecipeScraper:
- Left comment about 'Generate Recipe' button that I mentioned above. 
- Uncomment the assertion that applies once decided. 

Fixed test for HomePage:
- HomePage doesn't actually have any behaviour. It's all in RecipeScraper. So I deleted most of the tests. 
- I hope refactoring this test would make our tests less prone to False Negatives. 

### ToDo / Things to consider:
- [ ] Need to test for the new elements added to HomePage
- [ ] All these tests were unit tests. How to write an Integration test when the states are all in App.js?

kawrou committed:
Fix: Authentication tests. Addition: checkToken tests

kawrou committed:
Added some comments for things to check

kawrou committed:
Test: Redid tests. Added new test cases. Added comments

kawrou committed:
Added comments about changes

kawrou committed:
Fix:HomePage test

gabrielaehrenbrink commented:
Yeah I believe the data in services isnt being used. I think the changes you've made seem to look good

ben-dh3 commented:
> CheckToken seemed integral to the functioning of RecipeScraper so I started with that and refactored the code a little bit. As CheckToken only requires that the controller sends a response code, I changed the controller to send 200 or 500. 401 is handled by tokenChecker. I got rid of the data in services as the data isn't used for anything. **Please check that I'm correct in that assumption**

Yep agree with this! :+1: Not sure what 'manual' prop is in handleclick function either

kawrou commented:
> > CheckToken seemed integral to the functioning of RecipeScraper so I started with that and refactored the code a little bit. As CheckToken only requires that the controller sends a response code, I changed the controller to send 200 or 500. 401 is handled by tokenChecker. I got rid of the data in services as the data isn't used for anything. **Please check that I'm correct in that assumption**
> 
> Yep agree with this! 👍 Not sure what 'manual' prop is in handleclick function either

It isn't a prop. I don't know who wrote that code so I was confused at first as well haha. It's value is a boolean from the buttons below. One of the buttons is false and the other is true.
